From 187588e67522b54429a05975396233ea2cf695a3 Mon Sep 17 00:00:00 2001
From: "Miss Islington (bot)"
 <31488909+miss-islington@users.noreply.github.com>
Date: Sat, 11 Oct 2025 19:59:55 +0200
Subject: [PATCH] [3.14] gh-139929: fix incorrect OpenSSL version-based guard
 in `_ssl.c` (GH-139945) (#139964)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gh-139929: fix incorrect OpenSSL version-based guard in `_ssl.c` (GH-139945)

fix OpenSSL version-based guards
(cherry picked from commit cdd3eee7fc26538c8365dcbf2dd844ec7cdf7fb7)

Co-authored-by: Bénédikt Tran <10796600+picnixz@users.noreply.github.com>
---
 Modules/_ssl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Modules/_ssl.c b/Modules/_ssl.c
index 6898d73b8ffb2a..051efce4cfaa99 100644
--- a/Modules/_ssl.c
+++ b/Modules/_ssl.c
@@ -910,7 +910,7 @@ newPySSLSocket(PySSLContext *sslctx, PySocketSockObject *sock,
     }
 
     /* bpo43522 and OpenSSL < 1.1.1l: copy hostflags manually */
-#if OPENSSL_VERSION < 0x101010cf
+#if OPENSSL_VERSION_NUMBER < 0x101010cf
     X509_VERIFY_PARAM *ssl_verification_params = SSL_get0_param(self->ssl);
     X509_VERIFY_PARAM *ssl_ctx_verification_params = SSL_CTX_get0_param(ctx);
 
